src/Input/Select/Select.test.tsx

```diff
@@ -360,3 +360,196 @@ it('up and down arrow keys can be used to navigate the options menu', () => {
+
+    const Component = () => (
+      <React.Fragment>
+        <Select label="ClickOutside">
+          <Select.Option value="ClickOutside">ClickOutside</Select.Option>
+        </Select>
+      </React.Fragment>
+    );
+
+    const { unmount } = render(<Component />);
+    unmount();
+    expect(removeEventListenerSpy.mock.calls[0][1]).toEqual(
+      addEventListenerSpy.mock.calls[0][1]
+    );
+    addEventListenerSpy.mockRestore();
+    removeEventListenerSpy.mockRestore();
+  });
+});
+
+describe('<Select/> onBlur onFocus', () => {
+  it('should call onBlur and onFocus', () => {
+    const onBlurSpy = jest.fn();
+    const onFocusSpy = jest.fn();
+    const Component = () => (
+      <React.Fragment>
+        <Select label="ClickOutside" onBlur={onBlurSpy} onFocus={onFocusSpy}>
+          <Select.Option value="ClickOutside">ClickOutside</Select.Option>
+        </Select>
+      </React.Fragment>
+    );
+
+    const { queryByRole, asFragment } = render(<Component />);
+    const selectInput = queryByRole('combobox') as HTMLInputElement;
+
+    fireEvent.focus(selectInput);
+    expect(onFocusSpy).toHaveBeenCalledTimes(1);
+    // use snapshot to validate isFocus
+    expect(asFragment()).toMatchSnapshot();
+
+    fireEvent.blur(selectInput);
+    expect(onBlurSpy).toHaveBeenCalledTimes(1);
+    // use snapshot to validate floating
+    expect(asFragment()).toMatchSnapshot();
+  });
+});
+
+describe('<Select/> handleValueChange', () => {
+  it('input value should be the same as passed prop value', async () => {
+    const Component = ({ value }: { value: string }) => (
+      <React.Fragment>
+        <Select label="handleValueChange" value={value}>
+          <Select.Option value="handleValueChange">
+            handleValueChange
+          </Select.Option>
+        </Select>
+      </React.Fragment>
+    );
+
+    const mockValue = 'mock';
+    const { queryByRole, rerender } = render(<Component value={mockValue} />);
+    const selectInput = queryByRole('combobox');
+    expect(selectInput.getAttribute('value')).toEqual(mockValue);
+    rerender(<Component value="" />);
+    expect(selectInput.getAttribute('value')).toEqual('');
+  });
+});
+
+describe('<Select/> prop onInputChange', () => {
+  it('should call onInputChange on input change', async () => {
+    const onInputChangeSpy = jest.fn();
+    const Component = () => (
+      <React.Fragment>
+        <Select label="onInputChange" onInputChange={onInputChangeSpy}>
+          <Select.Option value="onInputChange">onInputChange</Select.Option>
+        </Select>
+      </React.Fragment>
+    );
+
+    const mockValue = 'mock value';
+    const { queryByRole } = render(<Component />);
+    const selectInput = queryByRole('combobox');
+    fireEvent.change(selectInput, {
+      target: {
+        value: mockValue,
+      },
+    });
+    expect(onInputChangeSpy).toHaveBeenCalledTimes(1);
+    // not able to validate SyntheticEvent
```

Comment:
What does this comment mean?

Reply:
we cannot validate the argument of `onInputChangeSpy` when it's being called, because the event object is a SyntheticEvent.